validator/project_validator.go

```diff
@@ -1572,6 +1573,25 @@ func validateTaskSyncSettings(p *model.Project, ref *model.ProjectRef) Validatio
 	return errs
 }
 
+// validateVersionControl checks if a project with defined project config fields has version control enabled on the project ref.
+func validateVersionControl(p *model.Project, ref *model.ProjectRef, config *model.ProjectConfig) ValidationErrors {
```

Comment:
Rather than passing in the whole config, Would it make sense to just pass in a boolean isConfigDefined or something?